### PR TITLE
Make styled dropdowns accessible DAH-124

### PR DESCRIPTION
**Pull request by akegan**
DAH-124

This PR:
- Switches our styled select component to use react-select
- Makes it so that you can use arrows to navigate our styled selects (currently just status and substatus dropdowns
- Updates the status modal wrapper to be a bit more inline with the rest of our form improvements

Still TODO:
- [x] Fix hover/focus/selected colors
- [x] Add tests
- [ ] Add example to the pattern lib

### CODE REVIEW

File: app/javascript/components/molecules/StatusDropdown.js
Comment (akegan) [inline app/javascript/components/molecules/StatusDropdown.js:40]: if all of our styled dropdowns are going to be buttons then it would make sense to make more generalized. The differences between substatus and status dropdowns are just a couple of classes

File: app/javascript/components/atoms/colors.js
Comment (akegan) [inline app/javascript/components/atoms/colors.js:4]: here's the colors constants I pulled in. figured we could generalize to styles if we need to at some point, but for now colors seem like enough

File: app/javascript/components/organisms/StatusModalWrapper.js
Comment (akegan) [inline app/javascript/components/organisms/StatusModalWrapper.js:91]: could be nice to pull these into Field.js but I think they are maybe just slightly too complicated and specific for now.

**Review by akegan — COMMENTED** [review]
Looks like just 2 items where I'm thinking I could refactor more.

File: app/javascript/components/molecules/SubstatusDropdown.js
Comment (james-wills-sf) [inline app/javascript/components/molecules/SubstatusDropdown.js:42]: what are the other props that can be passed in here? Is it just a couple that can be named instead?

File: spec/javascript/components/organisms/StatusModalWrapper.test.js
Comment (james-wills-sf) [inline spec/javascript/components/organisms/StatusModalWrapper.test.js:14]: I assume we need to import with `require` because of jest? Can you make a comment explaining that?

File: spec/javascript/components/organisms/StatusModalWrapper.test.js
Comment (james-wills-sf) [inline spec/javascript/components/organisms/StatusModalWrapper.test.js]: I think when it's a component that's already exported it's better to import and pass in the component itself instead of using the string name

```
import SubstatusDropdown from '~/components/molecules/SubstatusDropdown'
...
expect(wrapper.find(SubstatusDropdown).props().hasError).toBe(true) 
```

This comment also goes for `TextAreaField`

File: spec/javascript/e2e/LeaseUpPage.e2e.js
Comment (james-wills-sf) [inline spec/javascript/e2e/LeaseUpPage.e2e.js]: be sure to remove the .only here!

**Review by james-wills-sf — APPROVED** [review]
@akegan QA'd and things look good!

One additional question I have is why is toolkit.scss modified?

File: app/javascript/components/molecules/SubstatusDropdown.js
Comment (akegan) [reply app/javascript/components/molecules/SubstatusDropdown.js:42]: I'm not sure, they're passed by Select and they don't necessarily enumerate them in the docs. I think there might be a lot though.

File: spec/javascript/components/molecules/StatusDropdown.test.js
Comment (james-wills-sf) [inline spec/javascript/components/molecules/StatusDropdown.test.js]: I think this might be tied a little closely to the implementation, since I think this will check ordering too.

How about we just check that the is-appealed class is added here, since that seems most relevant to the test

File: spec/javascript/components/molecules/StatusDropdown.test.js
Comment (james-wills-sf) [inline spec/javascript/components/molecules/StatusDropdown.test.js]: I think semantically you're naming some of these tests more like how a describe block would be named.

I think `test` blocks should be more like "should do X" or "does X when Y"

So this could be test('renders a placeholder when provided')

File: spec/javascript/components/molecules/StatusDropdown.test.js
Comment (james-wills-sf) [inline spec/javascript/components/molecules/StatusDropdown.test.js]: Another case where I'm concerned that we're checking ordering where we don't really care what order the classes are in. Could this be two separate expects, one for each class we want to check?

**Review by james-wills-sf — COMMENTED** [review]
@akegan thanks for writing all those tests, lgtm but I'll re-review after the merge conflicts are resolved

File: spec/javascript/components/molecules/StatusDropdown.test.js
Comment (akegan) [reply spec/javascript/components/molecules/StatusDropdown.test.js]: updated in StatusDropdown and SubstatusDropdown

**Comment by akegan** [comment]
@james-wills-sf OK merge conflicts resolved and comments addressed! Just needs a last quick review + QA then we should hopefully be able to merge

File: spec/javascript/components/molecules/Dropdown.test.js
Comment (james-wills-sf) [inline spec/javascript/components/molecules/Dropdown.test.js:6]: Thanks for adding a comment here!

**Review by james-wills-sf — APPROVED** [review]

**Pull request merged**
